x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFormatProcessorTests.java

```diff
@@ -57,14 +57,16 @@ protected ZoneId instanceZoneId(DateTimeFormatProcessor instance) {
 
     @Override
     protected DateTimeFormatProcessor mutateInstance(DateTimeFormatProcessor instance) {
+        Formatter replaced = randomValueOtherThan(instance.formatter(), () -> randomFrom(Formatter.values()));
         return new DateTimeFormatProcessor(
             new ConstantProcessor(DateTimeTestUtils.nowWithMillisResolution()),
             new ConstantProcessor(ESTestCase.randomRealisticUnicodeOfLength(128)),
-            randomValueOtherThan(instance.zoneId(), ESTestCase::randomZone)
+            randomZone(),
+            replaced
```

Comment:
nit: you could avoid the variable and use the `randomValueOtherThan()` here directly.

Reply:
updated